### PR TITLE
feat: add architecture patterns

### PR DESCRIPTION
**Pull request by skobelina**
- Transferred logic of the trigger with the new exchange rate  to the EmailSender service.
- Implemented functionality of unsubscribing
- Implemented event-driven approach and asynchronous communication
- Added consumer logic tests.

### CODE REVIEW

File: email_sender/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/skobelina/email_sender/internal/mails"
+	"github.com/skobelina/email_sender/internal/mails/templates"
+	"github.com/skobelina/email_sender/pkg/queue"
+)
+
+var (
+	rabbitMQURL = os.Getenv("RABBITMQ_URL")
```
Comment (KushnerykPavel) [inline email_sender/cmd/main.go]: can we move it to main function or create module which work with env/config variables ?

Comment (skobelina) [reply email_sender/cmd/main.go]: yes, fixed

File: infrastructure/currencies/currency-beacon.go [hunk 2]
```diff
@@ -31,14 +31,10 @@ type CurrencyBeaconDataResponse struct {
...
-var (
-	currencyBeaconApi = os.Getenv("APP_CURRENCY_BEACON_URL")
-	apiKeyBeacon      = os.Getenv("APP_CURRENCY_BEACON_KEY")
-)
-
 func (p *ProviderCurrencyBeacon) Handle() (float64, error) {
+	var config configs.Config
 	var myClient = &http.Client{Timeout: 10 * time.Second}
-	req, err := http.NewRequest("GET", currencyBeaconApi+"?api_key="+apiKeyBeacon+"&base=USD&symbols=UAH", nil)
+	req, err := http.NewRequest("GET", config.AppCurrencyBeaconURL+"?api_key="+config.AppCurrencyBeaconKey+"&base=USD&symbols=UAH", nil)
```
Comment (KushnerykPavel) [inline infrastructure/currencies/currency-beacon.go]:
```suggestion
	req, err := http.NewRequest(http.MethodGet, config.AppCurrencyBeaconURL+"?api_key="+config.AppCurrencyBeaconKey+"&base=USD&symbols=UAH", http.NoBody)
```

File: email_sender/pkg/utils/serializer/json.go [hunk 3]
```diff
@@ -164,3 +151,52 @@ func (e *ForbiddenError) Error() string {
...
+type JsonMessage struct {
+	Message string `json:"message"`
+}
+
+func SendJSON(w http.ResponseWriter, status int, object interface{}) error {
+	// set headers
+	SetCorsHeaders(w)
+	w.Header().Set("Content-Type", "application/json")
+	// write response
+	b, err := json.Marshal(object)
```
Comment (KushnerykPavel) [inline email_sender/pkg/utils/serializer/json.go]: can we try to use here json.NewEncoder(w).Encode(object)

Comment (skobelina) [reply email_sender/pkg/utils/serializer/json.go]: yes, that's right, thank you

**Pull request merged**